.github/workflows/push-foundation-repo.yml

```diff
@@ -2,16 +2,79 @@ name: Sync Cardinal-Cryptography repo with Aleph-Zero-Foundation repo
 
 on:
   push:
-    branches: [main]
+    branches:
+      - main
+      - release-**
+    tags:
+      - r-*
 
 jobs:
-  sync:
+  sync-main:
     runs-on: ubuntu-20.04
-    if: ${{ github.repository == 'Cardinal-Cryptography/aleph-node'}}
+    if: github.repository == 'Cardinal-Cryptography/aleph-node' && startsWith(github.ref, 'refs/heads/') && github.ref_name == 'main'
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v2
         with:
           fetch-depth: 0
           token: ${{ secrets.SYNCAZF }}
+
       - name: Push to Aleph-Zero-Foundation
         run: git push https://x-access-token:${{ secrets.SYNCAZF }}@github.com/aleph-zero-foundation/aleph-node.git
+
+
+  sync-release-branch:
+    runs-on: ubuntu-20.04
+    if: github.repository == 'Cardinal-Cryptography/aleph-node' && startsWith(github.ref, 'refs/heads/') && startsWith(github.ref_name, 'release-')
+    steps:
+      - name: GIT | Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.SYNCAZF }}
+
+      - name: Get branch name and commit SHA
+        id: get_branch
+        shell: bash
+        env:
+          HEAD_REF: ${{ github.ref }}
+        run: |
+          echo branch_name=$(echo ${HEAD_REF#refs/heads/}) >> $GITHUB_OUTPUT
+          echo sha_short=$(git rev-parse --short HEAD) >> $GITHUB_OUTPUT
+
+      - name: Push to Aleph-Zero-Foundation
+        run: |
+          git remote add mirror 'https://x-access-token:${{ secrets.SYNCAZF }}@github.com/aleph-zero-foundation/aleph-node.git'
```

Comment:
Add mirror is only needed for branches different than main? Asking since it's different than for main.

Reply:
I have checked it again.  It actually works without adding origin.  My bad.  It wasn't working for me earlier, but it was a different issue - not related to adding that origin.